File: problems/array-sum/array_sum.php
<?php

function arraySum($arr)
{
  $sum = 0;
  foreach ($arr as $item)
  {
    if (is_array($item))
    {
      $sum += arraySum($item);
    }
    else
    {
      $sum += $item;
    }
  }
  return $sum;
}

var_dump(arraySum([1,2,[3,4,[5]]]));

File: problems/array-sum/array-sum.java
import java.util.Arrays;
import java.util.stream.Stream;

public class ArraySum {

    int arraySum(Object[] arr) {
        return flatten(arr).mapToInt(o -> (int) o).sum();
    }

    private Stream<Object> flatten(Object[] array) {
        return Arrays.stream(array)
            .flatMap(o -> o instanceof Object[] ? flatten((Object[]) o) : Stream.of(o));
    }
}
